### ISSUE TITLE
Where can I get the yolo.h5 file?

### ISSUE THREAD

**Issue by miranthajayatilake** [JudasDie/deeplearning.ai#2]
Great content. I was following the car detection notebook in the Week3 of Convolutional Neural Networks. I notices that the pretrained yolo.h5 file is missing from the directory. Is there any place I can download this from?

Thanks in advance.

**Comment by miranthajayatilake** [comment]
So I found the solution:

The yolo.h5 file can be generated using the YAD2K repository here: https://github.com/allanzelener/YAD2K

Steps how to do it on windows:
- Clone the above repository to your computer
- Download the yolo.weights file from here: http://pjreddie.com/media/files/yolo.weights
- Download the yolo.cfg file form here: https://raw.githubusercontent.com/pjreddie/darknet/master/cfg/yolo.cfg

- Copy the downloaded weights and cfg files to the YAD2K master directory

Run `python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5` on the shell and the h5 file will be generated.

**Comment by dineshmane** [comment]
Thanks it helped 👍

**Comment by HanYuanyuaner** [comment]
@miranthajayatilake the link for yolo.cfg not worked.

**Comment by ShawnZhong** [comment]
Use this one:
https://raw.githubusercontent.com/pjreddie/darknet/master/cfg/yolov1.cfg

**Comment by Estapraq** [comment]
Just go to Yolo website and download it 
https://pjreddie.com/darknet/yolo/

**Comment by anubhav0fnu** [comment]
> * https://raw.githubusercontent.com/pjreddie/darknet/master/cfg/

@miranthajayatilake , I am not able to download the configuration file?  the link isn't valid! Also, does the Yolo weight corresponds to version2 or version 3?

**Comment by msomeha** [comment]
You can get the weights and cfg file from this website.
https://pjreddie.com/darknet/yolo/

The coursera course is using YOLOv2 608 x 608.  You will need tensoflow v1 to compile.

**Comment by Sameedquais** [comment]
> So I found the solution:
> 
> The yolo.h5 file can be generated using the YAD2K repository here: https://github.com/allanzelener/YAD2K
> 
> Steps how to do it on windows:
> 
> * Clone the above repository to your computer
> * Download the yolo.weights file from here: http://pjreddie.com/media/files/yolo.weights
> * Download the yolo.cfg file form here: https://raw.githubusercontent.com/pjreddie/darknet/master/cfg/yolo.cfg
> * Copy the downloaded weights and cfg files to the YAD2K master directory
> 
> Run `python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5` on the shell and the h5 file will be generated.

**Comment by Sameedquais** [comment]
Can anyone explain this step. I have downloaded both cfg and weight file. I don't understand where and how to run it?
> Run `python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5` on the shell and the h5 file will be generated.

**Comment by XueminLiu111** [comment]
After run **python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5** on the shell , I got another bug:
**ModuleNotFoundError: No module named 'keras_yolo'**
Anyone know how to solve it?
Thx

**Comment by iamsurya** [comment]
Folks, this code is from deeplearning.ai/ and you can sign up for their free course and work this out.

**Comment by creative-h** [comment]
> Can anyone explain this step. I have downloaded both cfg and weight file. I don't understand where and how to run it?
> 
> > Run `python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5` on the shell and the h5 file will be generated.
@Sameedquais 
you should run in command prompt or Anaconda prompt with same directory as YAD2K master.

**Comment by geserit5** [comment]
I get a error message when I run yad2k.py
"Traceback (most recent call last):
  File "yad2k.py", line 270, in <module>
    _main(parser.parse_args())
  File "yad2k.py", line 156, in _main
    buffer=weights_file.read(weights_size * 4))
TypeError: buffer is too small for requested array "
Any suggestions?

**Comment by VEDANTKHARE-JEC** [comment]
how can i run it in google colab?

**Comment by kaushil24** [comment]
> how can i run it in google colab?

1. Clone repo using `!git clone  https://github.com/allanzelener/YAD2K.git`
2. Go inside the directory using `os.chdir(folder_name)`
2. Get the weights with:
```
!wget https://pjreddie.com/media/files/yolov2-voc.weights
!wget https://raw.githubusercontent.com/pjreddie/darknet/master/cfg/yolov2-voc.cfg
```
3. Run `!python3 yad2k.py yolov2-voc.cfg yolov2-voc.weights model_data/yolov2-voc.h5`

---
Tip: To run any terminal command on colab you can use `!` before the command. [Refer this](https://colab.research.google.com/notebooks/basic_features_overview.ipynb#scrollTo=Wej_mEyXQSHc)

**Comment by Mr-Zhu-pg** [comment]
> I get a error message when I run yad2k.py
> "Traceback (most recent call last):
> File "yad2k.py", line 270, in
> _main(parser.parse_args())
> File "yad2k.py", line 156, in _main
> buffer=weights_file.read(weights_size * 4))
> TypeError: buffer is too small for requested array "
> Any suggestions?

Hi, we must make yolo.cfg and yolo.weights have the same version, yolo having many versions like yolo-tiny,yolov1,yolov2,yolov3....

**Comment by galla-rupesh** [comment]
> You can get the weights and cfg file from this website.
> https://pjreddie.com/darknet/yolo/
> 
> The coursera course is using YOLOv2 608 x 608. You will need tensoflow v1 to compile.

Hi, but to degrade tensorflow to v1, yolo.utils is using keras which needs tf v>2.0 then how to proceed? and while using yolov2 h5 file in tf version 2.4.0, its giving error "yad2k.models.keras_yolo is not loaded, but a Lambda layer uses it. It may cause errors." while loading model. any suggestions please?

**Comment by MasoumehVahedi** [comment]
> Can anyone explain this step. I have downloaded both cfg and weight file. I don't understand where and how to run it?
> 
> > Run `python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5` on the shell and the h5 file will be generated.
Could you help me how to copy cfg and weigth file on master directory in google colab? I got error when I want to run `python yad2k.py yolo.cfg yolo.weights model_data/yolo.h5` on the shell and the h5

**Comment by NCcoco** [comment]
thank you

**Comment by Akhil-Tony** [comment]
anybody have a pre trained yolo 1 weight.h5 file ?

**Comment by khaledhasanzami** [comment]
> > I get a error message when I run yad2k.py
> > "Traceback (most recent call last):
> > File "yad2k.py", line 270, in
> > _main(parser.parse_args())
> > File "yad2k.py", line 156, in _main
> > buffer=weights_file.read(weights_size * 4))
> > TypeError: buffer is too small for requested array "
> > Any suggestions?
> 
> Hi, we must make yolo.cfg and yolo.weights have the same version, yolo having many versions like yolo-tiny,yolov1,yolov2,yolov3....

I have the same version, it is still showing this error

**Comment by parsa-pico** [comment]
for anyone who still has this problem in tensorflow 2,go to original website of yolov2 and download darknet and weights file,find yolo config in darknet folder, then using yad2k you can easily generate .h5 file,(when running yad2k.py you get bunch of import errors that you have to manually correct them because of new version of tensorflow)
yolo v2site:
https://pjreddie.com/darknet/yolov2/
installing yad2k:
git clone https://github.com/allanzelener/yad2k.git
put config and weight file inside yad2k folder
if you get space_to_depth err, go to the corresponding line and change it to tf.nn.space_to_depth

**Comment by tusharsinghsoam** [comment]
I am getting the following error:


Loading weights.
Weights Header:  [       0        1        0 32013312]
Parsing Darknet config.
Traceback (most recent call last):
  File "C:\Users\tsoam\Downloads\YAD2K\yad2k.py", line 274, in <module>
    _main(parser.parse_args())
  File "C:\Users\tsoam\Downloads\YAD2K\yad2k.py", line 95, in _main
    cfg_parser.read_file(unique_config_file)
  File "C:\Users\tsoam\AppData\Local\Programs\Python\Python311\Lib\configparser.py", line 733, in read_file
    self._read(f, source)
  File "C:\Users\tsoam\AppData\Local\Programs\Python\Python311\Lib\configparser.py", line 1100, in _read
    raise MissingSectionHeaderError(fpname, lineno, line)
configparser.MissingSectionHeaderError: File contains no section headers.


Note: I have already manually corrected import statements as I am using tf 2.13, Python version 3.11.

**Comment by tusharsinghsoam** [comment]
> I am getting the following error:
> 
> Loading weights. Weights Header: [ 0 1 0 32013312] Parsing Darknet config. Traceback (most recent call last): File "C:\Users\tsoam\Downloads\YAD2K\yad2k.py", line 274, in _main(parser.parse_args()) File "C:\Users\tsoam\Downloads\YAD2K\yad2k.py", line 95, in _main cfg_parser.read_file(unique_config_file) File "C:\Users\tsoam\AppData\Local\Programs\Python\Python311\Lib\configparser.py", line 733, in read_file self._read(f, source) File "C:\Users\tsoam\AppData\Local\Programs\Python\Python311\Lib\configparser.py", line 1100, in _read raise MissingSectionHeaderError(fpname, lineno, line) configparser.MissingSectionHeaderError: File contains no section headers.
> 
> Note: I have already manually corrected import statements as I am using tf 2.13, Python version 3.11.

I solved this by directly downloading yolov2.cfg from GitHub repo rather than using curl. I think curl messes up the format of the file.
